Sources/VaporJWT/Provider.swift

```diff
@@ -0,0 +1,104 @@
+import Vapor
+import JWT
+
+/// Adds required JWT objects to your application
+/// like token Signers
+public final class Provider: Vapor.Provider {
+    public let signer: Signer
+    public init(signer: Signer) {
+        self.signer = signer
+    }
+
+    /// Parses a `jwt.json` config file to create
+    /// the JWT objects
+    public convenience init(config: Config) throws {
+        guard let jwt = config["jwt"]?.object else {
+            throw ConfigError.noJWTConfigFile()
+        }
+
+        guard let signer = jwt["signer"]?.object else {
+            throw ConfigError.jwtConfigMissingRequiredKey(key: "signer", type: Dictionary<String, Any>.self)
+        }
+
+        guard let signerType = signer["type"]?.string else {
+            throw ConfigError.jwtConfigMissingRequiredKey(key: "signer.type", type: String.self)
+
+        }
+
+        switch signerType.lowercased() {
+        case "unsigned":
+            self.init(signer: Unsigned())
```

Comment:
Agreed. I've refactored this part a bit. Will push with the rest of the changes.